### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by asutula**
Update to Mobile.framework that starts up faster
Clean up sagas
Block UI until node is started

### CODE REVIEW

File: App/Redux/IpfsNodeRedux.js [hunk 1]
```diff
@@ -4,10 +4,7 @@ import Immutable from 'seamless-immutable'
 /* ------------- Types and Action Creators ------------- */
 
 const { Types, Creators } = createActions({
-  createNodeRequest: ['path'],
```
Comment (asutula) [inline App/Redux/IpfsNodeRedux.js]: Creating and starting the node are one in the same, so just combine them into `start`

**Comment by carsonfarmer** [comment]
Awesome. Will defs test this later!

File: App/Sagas/TextileSagas.js [hunk 2]
```diff
@@ -71,59 +71,79 @@ export function * recoverPassword ({data}) {
...
+    yield * triggerStopNode()
   }
+  yield call(BackgroundTimer.stop)
 }
 
-export function * startNode () {
+export function * startNode ({path}) {
+  const onboarded = yield select(TextileSelectors.onboarded)
+  if (!onboarded) {
+    return
```
Comment (sanderpick) [inline App/Sagas/TextileSagas.js]: nice

File: App/Sagas/TextileSagas.js [hunk 3]
```diff
@@ -71,59 +71,79 @@ export function * recoverPassword ({data}) {
...
+    yield call(BackgroundTimer.stop)
   }
 }
 
-export function * stopNode ({newState}) {
-  if (newState !== 'inactive') {
-    return
-  }
+export function * stopNode () {
+  yield call(BackgroundTimer.start)
```
Comment (sanderpick) [inline App/Sagas/TextileSagas.js]: just checking, can this be scheduled by the OS to happen at some later date, or does it necessarily happen right away? if the former case is true, hopefully it's canceled before the node starts again

File: App/Sagas/index.js [hunk 4]
```diff
@@ -39,19 +39,25 @@ export default function * root () {
...
-
     takeEvery(IpfsNodeTypes.GET_PHOTO_HASHES_REQUEST, getPhotoHashes),
 
     takeEvery(UITypes.SHARE_PHOTO_REQUEST, shareImage),
 
-    takeEvery(TextileTypes.APP_STATE_CHANGE, photosTask),
-    takeEvery(TextileTypes.LOCATION_UPDATE, photosTask),
-    takeEvery(TextileTypes.BACKGROUND_TASK, photosTask),
+    // Actions that trigger starting/stopping the node
+    takeEvery(action => action.type === TextileTypes.APP_STATE_CHANGE && action.newState !== 'background', handleStateChange),
```
Comment (sanderpick) [inline App/Sagas/index.js]: sweet

File: App/Sagas/TextileSagas.js (hunk 3 above)
Comment (asutula) [reply App/Sagas/TextileSagas.js]: man, good question. it's on my list to understand how to compose all these flows in a way where things are canceled etc as needed, but that's some complicated stuff. this should happen right away, but it seems possible that the app could be foregrounded again or receive a location update before the node is done stopping. It's a complicated dependency graph of things

**Comment by andrewxhill** [comment]
you ever see this on this branch?

screen shot 2018-05-08 at 6 35 31 pm

this is with a valid name/pass

**Comment by asutula** [comment]
No, but don't think i've tried to log in, only sign up. Trying now.

**Comment by asutula** [comment]
Log in worked fine on iOS

File: App/Sagas/TextileSagas.js (hunk 3 above)
Comment (sanderpick) [reply App/Sagas/TextileSagas.js]: yeah, tricky stuff. good thing the node shuts down almost immediately

**Comment by andrewxhill** [comment]
okay, but login is broke in android in this branch. 

just merged it into my branch and will try to fix

**Comment by andrewxhill** [comment]
appears it doesn't start in android any more. calling a test here and get `isSignedIn() on null object reference`

**Comment by andrewxhill** [comment]
seems that without the Onboarded flag the node doesn't start. But login happens before onboarding.

**Comment by andrewxhill** [comment]
see my coments, pretty sure signin is broke in android. going to merge and we can fix from master

**Pull request merged**